contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol

```diff
@@ -0,0 +1,195 @@
+     * @param _expiry Time until dividend can no longer be paid, and can be reclaimed by issuer
+     */
+    function createDividendWithCheckpoint(uint256 _maturity, uint256 _expiry, address _token, uint256 _amount, uint256 _checkpointId) payable public onlyOwner {
+        require(_expiry > _maturity);
+        require(_checkpointId <= ISecurityToken(securityToken).currentCheckpointId());
+        uint256 dividendIndex = dividends.length;
+        uint256 currentSupply = ISecurityToken(securityToken).totalSupplyAt(_checkpointId);
+        require(ERC20(_token).transferFrom(msg.sender, address(this), _amount), "Unable to transfer tokens for dividend");
+        dividends.push(
+          Dividend(
+            _checkpointId,
+            now,
+            _maturity,
+            _expiry,
+            _token,
+            _amount,
+            0,
+            currentSupply,
+            false
+          )
+        );
+        emit ERC20DividendDeposited(msg.sender, _checkpointId, now, _maturity, _expiry, _token, _amount, currentSupply, dividendIndex);
+    }
+
+    /**
+     * @dev Issuer can push dividends to provided addresses
+     * @param _dividendIndex Dividend to push
+     * @param _payees Addresses to which to push the dividend
+     */
+    function pushDividendPaymentToAddresses(uint256 _dividendIndex, address[] _payees) public onlyOwner validDividendIndex(_dividendIndex) {
+        Dividend storage dividend = dividends[_dividendIndex];
+        for (uint256 i = 0; i < _payees.length; i++) {
+            if (!dividend.claimed[_payees[i]]) {
+                _payDividend(_payees[i], dividend, _dividendIndex);
+            }
+        }
+    }
+
+    /**
+     * @dev Issuer can push dividends using the investor list from the security token
+     * @param _dividendIndex Dividend to push
+     * @param _start Index in investor list at which to start pushing dividends
+     * @param _iterations Number of addresses to push dividends for
+     */
+    function pushDividendPayment(uint256 _dividendIndex, uint256 _start, uint256 _iterations) public onlyOwner validDividendIndex(_dividendIndex) {
+        Dividend storage dividend = dividends[_dividendIndex];
+        uint256 numberInvestors = ISecurityToken(securityToken).getInvestorsLength();
+        for (uint256 i = _start; i < Math.min256(numberInvestors, _start.add(_iterations)); i++) {
+            address payee = ISecurityToken(securityToken).investors(i);
+            if (!dividend.claimed[payee]) {
+                _payDividend(payee, dividend, _dividendIndex);
+            }
+        }
+    }
+
+    /**
+     * @dev Investors can pull their own dividends
+     * @param _dividendIndex Dividend to pull
+     */
+    function pullDividendPayment(uint256 _dividendIndex) public validDividendIndex(_dividendIndex)
+    {
+        Dividend storage dividend = dividends[_dividendIndex];
+        require(dividend.claimed[msg.sender] == false);
```

Comment:
Change to require (!dividend.claimed[msg.sender]) to keep consistency

Reply:
done & added errror message